modules/dnn/src/op_inf_engine.cpp

```diff
@@ -545,6 +545,18 @@ void InfEngineBackendNet::initPlugin(InferenceEngine::ICNNNetwork& net)
         }
         plugin = InferenceEngine::InferencePlugin(enginePtr);
 
+        InferenceEngine::ICNNNetwork::InputShapes inShapes = netOwner.getInputShapes();
+        InferenceEngine::ICNNNetwork::InputShapes::iterator itr;
+        for (itr = inShapes.begin(); itr != inShapes.end(); ++itr)
+        {
+            if(inpBlobs.find(itr->first) == inpBlobs.end())
+            {
+                CV_Error(Error::StsAssert, "BlobMap key not exist in Blob");
+            }
+            itr->second[2] = inpBlobs[itr->first]->dims()[1];
+            itr->second[3] = inpBlobs[itr->first]->dims()[0];
+        }
+        netOwner.reshape(inShapes);
```

Comment:
`netOwner` can be uninitialized in case of networks which are built in runtime (aren't loaded from `.xml` and `.bin` but from origin frameworks format). Please check `hasNetOwner` flag.